content/en/docs/reference/kubernetes-api/ref/common-definitions/objectmeta-.md

```diff
@@ -0,0 +1,182 @@
+---
+api_metadata:
+  apiVersion: ""
+  import: "k8s.io/apimachinery/pkg/apis/meta/v1"
+  kind: "ObjectMeta"
+collapsible: false
+content_type: "api_reference"
+description: "ObjectMeta is metadata that all persisted resources must have, which includes all objects users must create."
+draft: false
+title: "ObjectMeta"
+weight: 13
+---
+
+
+
+`import "k8s.io/apimachinery/pkg/apis/meta/v1"`
+
+
+## ObjectMeta
+
+ObjectMeta is metadata that all persisted resources must have, which includes all objects users must create.
+
+
+- **name** (string)
+
+  Name must be unique within a namespace. Is required when creating resources, although some resources may allow a client to request the generation of an appropriate name automatically. Name is primarily intended for creation idempotence and configuration definition. Cannot be updated. More info: http://kubernetes.io/docs/user-guide/identifiers#names
+
+- **generateName** (string)
+
+  GenerateName is an optional prefix, used by the server, to generate a unique name ONLY IF the Name field has not been provided. If this field is used, the name returned to the client will be different than the name passed. This value will also be combined with a unique suffix. The provided value has the same validation rules as the Name field, and may be truncated by the length of the suffix required to make the value unique on the server.
+  
+  If this field is specified and the generated name exists, the server will NOT return a 409 - instead, it will either return 201 Created or 500 with Reason ServerTimeout indicating a unique name could not be found in the time allotted, and the client should retry (optionally after the time indicated in the Retry-After header).
+  
+  Applied only if Name is not specified. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#idempotency
+
+- **namespace** (string)
+
+  Namespace defines the space within which each name must be unique. An empty namespace is equivalent to the "default" namespace, but "default" is the canonical representation. Not all objects are required to be scoped to a namespace - the value of this field for those objects will be empty.
+  
+  Must be a DNS_LABEL. Cannot be updated. More info: http://kubernetes.io/docs/user-guide/namespaces
+
+- **labels** (map[string]string)
+
+  Map of string keys and values that can be used to organize and categorize (scope and select) objects. May match selectors of replication controllers and services. More info: http://kubernetes.io/docs/user-guide/labels
+
+- **annotations** (map[string]string)
+
+  Annotations is an unstructured key value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata. They are not queryable and should be preserved when modifying objects. More info: http://kubernetes.io/docs/user-guide/annotations
+
+
+
+### System
```

Comment:
This "System" is confusing. Where does it come from? And ... what does it mean to users?

Reply:
The "System" category name is configured here: https://github.com/feloy/kubernetes-resources-reference/blob/master/config/v1.19/fields.yaml#L668

For each Definition, the generator gives a chance to group the fields by categories, and order the fields. 

I tried to separate the fields that are set by the user/developer (name, generatedName, namespace, labels, annotations) and the fields set by some controller (finalizers, managedFields, ownerReferences), but there is room for improvement.